### DISCUSSION TITLE
PGR Service V2 Deployment - jsons

### DISCUSSION THREAD

**Discussion by atulgupta2024** [egovernments/Digit-Core#314] [DIGIT Platform]
Am referring to the PGR Service 2.0 integration - https://digit-discuss.atlassian.net/wiki/spaces/DD/pages/2361655305/Integrating+PGR+In+mGramSeva

I can see the various JSON files that need to be modified. Is there a place where can find the details of schema of JSON file i.e. what each field is used for. Some fields are self explanatory, but not all of them. For example in 

- ServiceDefs.json, what is "Keywords" used for? 
1. Am assuming serviceCode is used only for internal/programmatic use. If yes, why is it so verbose as the "name" field? using a shorter code should be possible, right? 
2. name field is probably what is used to display on the UI 
3. What would active = false mean? Not display on the UI or show, but disabled? Why even display a disabled item? Is this changed via code during execution, based on roles? 
4. What are the various keywords used for? 
5. Is the department here expected to match up against a department defined elsewhere? 
6. The order field is not mentioned in all items in the json. Why so? and what does this order signify? Sequence in which items appear in the menu on the UI? Why not just appear as sorted alphabetically as that's how most menu items appear, unless they are grouped as per functionality

- What is complainclosingtime.json used for? Is the unit in milli seconds? How is this different from slahours in Service.Defs? 
- What is UIConstants.json for? 
- There is a separate tenants.json file and there are tenants defined in citymodule.json. What's the difference?

**Comment by jagankumar-egov** [comment]
Hi Atul,

Please refer the following for those queries , 
ServiceDefs.json, what is "Keywords" used for? Keywords are being used in the chatbot to pick the complaint type based on any keyword.
1.   Certainly, We use ServiceCode field
 and localize it. The name field serves solely to identify what this master is, preventing duplication by others.
2. We don’t show name field in UI
3.  Yes just to disable it in UI.
4.  Keywords are based on the complaint type,
5. Depending on the department, the employee for taking action varies. Employees also get assigned to a department.
6. Based on order field UI shows them the complaint list.

tenants file to know more about each tenant and city module to enable particular modules in tenant

**Comment by atulgupta2024** [comment]
1. Name is used to prevent duplicates or ServiceCode is? From the file, I have, it seems the name field is what is shown in the complaint "sub category" (after localization). The serviceCode probably is the key used to set the localized Text and also for uniqueness 
2. it a field is disabled always, because active=false, why even show it in the UI. I mean we can just remove the entire entry? Unless, there is a way to, at runtime, change it to active=true? 
3. There is department that is shown in HRMS. So there should be a department master that HRMS uses. Are we referring to the same departments? If yes then we should just use the ID here and not the actual department name
4. so the order is within the particular category? also, in case order is not defined, then is it the order in which the items appear in the file, or it is alphabetical? If order is not defined, I would prefer that the system show the fields in alphabetical order.

**Reply by jagankumar-egov** [reply]
1. service code is always unique, name is just to identify the master,
2. there might be some data referring to that code, which we cant remove completely so we make inactive.
3. Both PGR and HRMS uses same department master, here we just refer that department code
4. we have to enhance/ customise to get this behaviour.

**Comment by atulgupta2024** [comment]
Thanks Jagan for the responses. In general the DIGIT documentation should be updated to provide explanation for each of the json field. Examples are usually not sufficient to help explain the purpose, the format etc. for each field. and this is not for just PGR, but across all services.
